build/lib/src/generate/expected_outputs.dart

```diff
@@ -147,7 +168,22 @@ class _CapturingBuildOutputs extends _ParsedBuildOutputs {
     final lengthOfMatch = match.end - match.start;
 
     return _outputs.map((output) {
-      final resolvedOutput = output.replaceFirst('{{}}', match.group(1)!);
+      final resolvedOutput = output.replaceAllMapped(
+        _ParsedBuildOutputs._captureGroup,
+        (outputMatch) {
+          final name = outputMatch.group(1)!;
+          final index = _groupNames.indexOf(name);
+
+          if (index < 0) {
+            // The output refers to a capture group that does not appear in the
```

Comment:
I think we should make it an error to have a capture group in outputs that doesn't match one from inputs. I don't think we should support `{{}}` as raw text in output extensions, its much more likely to just be a typo. 

If somebody really needs that we could allow escaping it later or something haha, but I highly doubt this would come up.